aws-secretsmanager/src/main/java/io/micronaut/aws/secretsmanager/SecretsManagerGroupNameAwareKeyValueFetcher.java

```diff
@@ -0,0 +1,187 @@
+import software.amazon.awssdk.services.secretsmanager.model.ListSecretsResponse;
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.secretsmanager.model.SecretListEntry;
+import software.amazon.awssdk.services.secretsmanager.model.SecretsManagerException;
+
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * {@link GroupNameAwareKeyValueFetcher} implementations for AWS Secrets Manager.
+ *
+ * @author sbodvanski
+ * @since 3.8.0
+ */
+@Experimental
+@Requires(beans = {SecretsManagerClient.class})
+@BootstrapContextCompatible
+@Singleton
+public class SecretsManagerGroupNameAwareKeyValueFetcher implements GroupNameAwareKeyValueFetcher {
+    private static final Logger LOG = LoggerFactory.getLogger(SecretsManagerGroupNameAwareKeyValueFetcher.class);
+
+    private final SecretsManagerClient secretsClient;
+    private final ObjectMapper objectMapper;
+
+    /**
+     * @param secretsClient Secrets Client
+     * @param objectMapper Object Mapper
+     */
+    public SecretsManagerGroupNameAwareKeyValueFetcher(SecretsManagerClient secretsClient,
+                                                       ObjectMapper objectMapper) {
+        this.secretsClient = secretsClient;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    @NonNull
+    public Optional<Map<String, Map>> keyValuesByPrefix(@NonNull String prefix) {
+        Map<String, Map> result = new HashMap<>();
+        try {
+            String nextToken = null;
+            do {
+                ListSecretsRequest.Builder builder = ListSecretsRequest.builder()
+                        .nextToken(nextToken)
+                        .filters(Filter.builder()
+                                .key(FilterNameStringType.NAME)
+                                .values(prefix)
+                                .build());
+                if (nextToken != null) {
+                    builder = builder.nextToken(nextToken);
+                }
+                ListSecretsRequest listSecretsRequest = builder.build();
+                ListSecretsResponse secretsResponse = secretsClient.listSecrets(listSecretsRequest);
+                List<SecretListEntry> secrets = secretsResponse.secretList();
+                if (LOG.isTraceEnabled()) {
+                    if (secrets.isEmpty()) {
+                        LOG.trace("zero secrets for prefix: {}", prefix);
+                    } else {
+                        LOG.trace("# {} secrets for prefix: {}", secrets.size(), prefix);
+                    }
+                }
+                for (SecretListEntry secret : secrets) {
+                    if (LOG.isTraceEnabled()) {
+                        LOG.trace("Evaluating secret {}", secret.name());
+                    }
+                    Map keyValues = new HashMap<>();
```

Comment:
Use a linked hashmap in case order is important

Reply:
The class which calls this method manages the order of precedence for all the active env prefixes with the LinkedHashMap. So here it shouldn't be important.